### ISSUE TITLE
realm: scatter/gather fails with more than 127 indirect instances

### ISSUE THREAD

**Issue by jpietarilagraham** [StanfordLegion/legion#906]

**Comment by streichler** [comment]
Temporary workaround for this is: 
```
diff --git a/runtime/realm/transfer/channel.cc b/runtime/realm/transfer/channel.cc
index 589e5c6..5fe877f 100644
--- a/runtime/realm/transfer/channel.cc
+++ b/runtime/realm/transfer/channel.cc
@@ -446,9 +446,9 @@ namespace Realm {
            update_bytes_read(input_control.control_port_idx,
                               icp.local_bytes_total, sizeof(unsigned));
            icp.local_bytes_total += sizeof(unsigned);
-           input_control.remaining_count = cword >> 8;
-           input_control.current_io_port = (cword & 0x7f) - 1;
-           input_control.eos_received = (cword & 128) != 0;
+           input_control.remaining_count = cword >> 12;
+           input_control.current_io_port = (cword & 0x7ff) - 1;
+           input_control.eos_received = (cword & 2048) != 0;
            log_xd.info() << "input control: xd=" << std::hex << guid << std::dec
                          << " port=" << input_control.current_io_port
                          << " count=" << input_control.remaining_count
@@ -480,9 +480,9 @@ namespace Realm {
                               ocp.local_bytes_total, sizeof(unsigned));
            ocp.local_bytes_total += sizeof(unsigned);
            assert(cword != 0);
-           output_control.remaining_count = cword >> 8;
-           output_control.current_io_port = (cword & 0x7f) - 1;
-           output_control.eos_received = (cword & 128) != 0;
+           output_control.remaining_count = cword >> 12;
+           output_control.current_io_port = (cword & 0x7ff) - 1;
+           output_control.eos_received = (cword & 2048) != 0;
            log_xd.info() << "output control: xd=" << std::hex << guid << std::dec
                          << " port=" << output_control.current_io_port
                          << " count=" << output_control.remaining_count
diff --git a/runtime/realm/transfer/transfer.cc b/runtime/realm/transfer/transfer.cc
index d53bc79..428412e 100644
--- a/runtime/realm/transfer/transfer.cc
+++ b/runtime/realm/transfer/transfer.cc
@@ -1768,8 +1768,9 @@ namespace Realm {
     // now try to write the control information
     if((output_count > 0) || input_done) {
-      unsigned cword = (((output_count * element_size) << 8) +
-                       (input_done ? 128 : 0) + // bit 7
+      assert(output_space_id < ((1 << 12) - 1));
+      unsigned cword = (((output_count * element_size) << 12) +
+                       (input_done ? 2048 : 0) + // bit 11
                        (output_space_id + 1));
       assert(cword != 0); 
```

**Comment by streichler** [comment]
This should be fixed with commit b78b2471.
